### PR TITLE
[chore] Add triaging processes and definitions to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

Adds some guidelines and definitions to help facilitate common understanding with the triage process.

cc @open-telemetry/collector-contrib-triagers

### CODE REVIEW

File: CONTRIBUTING.md
Comment (atoulme) [inline CONTRIBUTING.md:185]:
```suggestion
is fit to be added to a component. Code owners can be pinged by adding a component label to an issue, or will be pinged by
```

File: CONTRIBUTING.md
Comment (atoulme) [inline CONTRIBUTING.md:217]: I think mentioning flaky test would help here too
```suggestion
| `enhancement`        | Something that isn't an advertised feature that would be useful to users or maintainers.                                                                                                                |
| `flaky test`        | A test unexpectedly failed during CI, showing that there is a problem with the code base that code owners may want to investigate.  |
```

File: CONTRIBUTING.md
Comment (atoulme) [inline CONTRIBUTING.md:221]: This is where I'm fuzzy, how do we validate? Do we need an ack from someone or can we go by our own judgement?

File: CONTRIBUTING.md
Comment (codeboten) [inline CONTRIBUTING.md:178]:
```suggestion
Contributors with [triager](https://github.com/open-telemetry/opentelemetry-collector-contrib/#contributing) permissions can help move issues along is through adding missing component labels,
```

File: CONTRIBUTING.md
Comment (codeboten) [inline CONTRIBUTING.md:179]:
```suggestion
which helps organize issues and triggers automations to notify code owners. They can also use their familiarity with the Collector and
```

File: CONTRIBUTING.md
Comment (codeboten) [inline CONTRIBUTING.md:188]:
```suggestion
is fit to be added to a component. Code owners can be pinged by adding a component label to an issue, or will be pinged by
repository automations when an issue is opened or becomes stale. Code owners may not have triager permissions on the repository,
so they can help triage through investigation and by participating in discussions. They can also help organize issues by
[adding labels via comments](#adding-labels-via-comments).
```
```suggestion
is fit to be added to a component. Code owners will be notified by repository automation when:
* a component label is added to an issue
* an issue is opened
* the issue becomes stale

Code owners may not have triager permissions on the repository,
so they can help triage through investigation and by participating in discussions. They can also help organize issues by
[adding labels via comments](#adding-labels-via-comments).
```

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md:222]: That's a good point. It's really by the judgement of the triager or code owner. Obviously changes can always arise that will change the status, but in general I think it should be used as a flag that the issue is ready for work. I'll try to update this description to offer more guidelines. It may also make sense to give this its own section.

**Review by Aneurysm9 — APPROVED** [review]

File: CONTRIBUTING.md
Comment (mx-psi) [inline CONTRIBUTING.md:217]: (not necessarily on this PR) a couple other labels we may want to define:
-  `release:blocker`
- `Vendor Specific Component`

**Review by mx-psi — APPROVED** [review]

File: CONTRIBUTING.md
Comment (jpkrohling) [inline CONTRIBUTING.md:173]: And the issue's reporter. They might be implicit in the "OpenTelemetry community members" part, but it's good to be explicit.

File: CONTRIBUTING.md
Comment (djaglowski) [inline CONTRIBUTING.md:178]: Wording is a little jumbled here

File: CONTRIBUTING.md
Comment (frzifus) [inline CONTRIBUTING.md:173]:
```suggestion
Determining the root cause of issues is a shared responsibility between those with triager permissions, code owners,
```

File: CONTRIBUTING.md
Comment (jpkrohling) [inline CONTRIBUTING.md:178]:
```suggestion
Contributors with [triager](https://github.com/open-telemetry/opentelemetry-collector-contrib/#contributing) permissions can help move issues along is through adding missing component labels,
```

I'm not a native speaker, but this sounds a bit weird to me.

**Review by frzifus — COMMENTED** [review]
Thanks @evan-bradley for taking care of it!

File: CONTRIBUTING.md
Comment (jpkrohling) [inline CONTRIBUTING.md:232]: a bug is rarely insignificant, especially to the person who spent the time reporting it. How about changing this to "low-priority bug"?

Comment (codeboten) [inline CONTRIBUTING.md:232]:
```suggestion
| `priority:p3`        | A technical improvement, insignificant bug, or other minor issue. Generally something that is considered a "nice to have."                                                                              |
```
```suggestion
| `priority:p3`        | A technical improvement, lower priority bug, or other minor issue. Generally something that is considered a "nice to have."                                                                              |
```

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md:218]: These are both good to have, I've added them. Let me know if you'd like any changes to be made to the explanations.

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md]: Looks like I left a few things over from a previous re-wording, thanks.

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md]: That's a good catch, I agree. Added.

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md]: Yeah, that doesn't read well. I've corrected it, thanks.

File: CONTRIBUTING.md
Comment (evan-bradley) [reply CONTRIBUTING.md]: Agreed. I wanted to indicate that the bug probably had a workaround (otherwise it would be a p2), but I'm not sure we should be that prescriptive. Amended.

**Review by andrzej-stencel — APPROVED** [review]
Love it, thank you Evan for this!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
